src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs

```diff
@@ -798,44 +798,31 @@ internal void AddCluster(ClusterResultItem ClusterResultItem)
                     index++;
                 }
             }
-
-            clusterConnections.ForEach(connection =>
+            using (var undoGroup = wsViewModel.Model.UndoRecorder.BeginActionGroup())
             {
-                // Connect the nodes
-                var sourceNode = clusterMapping[connection.StartNode.NodeId];
-                var targetNode = clusterMapping[connection.EndNode.NodeId];
-                // The port index is 1- based (currently a hack and not expected from service)
-                var sourcePort = sourceNode.OutPorts.FirstOrDefault(p => p.Index == connection.StartNode.PortIndex - 1);
-                var targetPort = targetNode.InPorts.FirstOrDefault(p => p.Index == connection.EndNode.PortIndex - 1);
-                var commands = new List<DynamoModel.ModelBasedRecordableCommand>
-                    {
-                        new DynamoModel.MakeConnectionCommand(sourceNode.GUID.ToString(), connection.StartNode.PortIndex - 1, PortType.Output, DynamoModel.MakeConnectionCommand.Mode.Begin),
-                        new DynamoModel.MakeConnectionCommand(targetNode.GUID.ToString(), connection.EndNode.PortIndex - 1, PortType.Input, DynamoModel.MakeConnectionCommand.Mode.End),
-                    };
-                commands.ForEach(c =>
+                clusterConnections.ForEach(connection =>
                 {
-                    try
+                    // Connect the nodes    
+                    var sourceNode = clusterMapping[connection.StartNode.NodeId];
+                    var targetNode = clusterMapping[connection.EndNode.NodeId];
+                    // The port index is 1- based (currently a hack and not expected from service)
+                    var sourcePort = sourceNode.OutPorts.FirstOrDefault(p => p.Index == connection.StartNode.PortIndex - 1);
+                    var targetPort = targetNode.InPorts.FirstOrDefault(p => p.Index == connection.EndNode.PortIndex - 1);
+
+                    var connector = ConnectorModel.Make(sourceNode, targetNode, connection.StartNode.PortIndex - 1, connection.EndNode.PortIndex - 1);
+                    if (connector != null)
```

Comment:
This check also become necessary since connector is not guaranteed to be created successfully, and passing null to the recorder has unexpected behavior